### PR TITLE
Upgrade to balenalib Debian 10 Buster for AMD64 platform

### PR DESCRIPTION
**Pull request by stickz**
* Upgrade to a supported Debian 10 image for a newer version of transmission-daemon.

* Install adoptopenjdk-8-hotspot due to dropped support for openjdk-7-jre-headless.

* Export JAVA_HOME environment variable so play framework knows where to look.

* Install the required zlib development bindings for gzip module support on nginx.

* Upgrade nginx from 1.14.0 to 1.14.2 for hotfix for GCC 8+ compiler support.

### CODE REVIEW

**Comment by stickz** [comment]
My motivation behind these changes was to fix a few issues with the unsupported aging base image.

- The outdated version of transmission-daemon was having trouble connecting with peers on newer torrent clients.
- The free and used disk space was not being retrieved from the node. This was causing breakage with torrent placements.

I tested these changes. Everything functions as intended. The web interface is actually a little bit faster with this upgrade.

**Comment by erindru** [comment]
Wow, im super surprised that people are still finding this software useful. Thanks so much for taking the time to dig into things and make improvements

**Comment by stickz** [comment]
These changes did not apply to latest `openseedbox/client:latest` on Docker. Some tweaking may be required so people can use these changes. I had to set `amd64.Dockerfile` for my `Dockerfile location` on the latest docker tag, to get my fork working.

**Comment by xabolcs** [comment]
Thanks for your contribution! :+1: 

Looks like TravisCI configuration needs a refresh.

Did you test your changes against `armv7hf` and `aarch64`? Did you notice, there are `Dockerfile.tpl` and `generate-dockerfile.sh`?

**Comment by stickz** [comment]
No, I only tested my changes against `amd64`. I was hoping it would just build this platform, but I guess not.

File: docker/amd64.Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM resin/amd64-debian:jessie
+FROM balenalib/amd64-debian-node
```
Comment (xabolcs) [inline docker/amd64.Dockerfile:1]: @stickz , why did you switched to Node stack?

```
$ docker run --rm -it balenalib/amd64-debian-node /bin/sh
Here are a few details about this Docker image (For more information please visit https://www.balena.io/docs/reference/base-images/base-images/): 
Architecture: Intel 64-bit (x86-64) 
OS: Debian Buster 
Variant: run variant 
Default variable(s): UDEV=off 
The following software stack is preinstalled: 
Node.js v15.4.0, Yarn v1.22.4 
Extra features: 
- Easy way to install packages with `install_packages <package-name>` command 
- Run anywhere with cross-build feature  (for ARM only) 
- Keep the container idling with `balena-idle` command 
- Show base image details with `balena-info` command
# 
```

Just asking. I want to refresh the Docker images.

Comment (stickz) [reply docker/amd64.Dockerfile:1]: I didn't see the `balenalib/amd64-debian` stack. This can be replaced when you refresh the docker images.

**Comment by xabolcs** [comment]
> These changes did not apply to `openseedbox/client:latest` on Docker.

Out now!

image

**Pull request merged**